pkg/cache/sql/informer/listoption_indexer.go

```diff
@@ -427,38 +534,179 @@ func (l *ListOptionIndexer) validateColumn(column string) error {
+		if err != nil {
 			return "", nil, err
 		}
+		clauses = append(clauses, newClause)
+		params = append(params, newParams...)
+	}
+	return strings.Join(clauses, " OR "), params, nil
+}
 
-		orWhereClause += fmt.Sprintf(`f."%s" %s ? ESCAPE '\'`, columnName, opString)
-		format := strictMatchFmt
+//type Filter struct {
+//	Field   []string
+//	Match   []string
+//	Op      Op: one of
+//	Partial bool
+//}
+
+// Possible ops from the k8s parser:
+// KEY = and == (same) VALUE
+// KEY != VALUE
+// KEY exists []  # ,KEY, => this filter
+// KEY ! []  # ,!KEY, => assert KEY doesn't exist
+// KEY in VALUES
+// KEY notin VALUES
+
+func (l *ListOptionIndexer) getFieldFilter(filter Filter) (string, []any, error) {
+	opString := ""
+	escapeString := ""
+	columnName := toColumnName(filter.Field)
+	if err := l.validateColumn(columnName); err != nil {
+		return "", nil, err
+	}
+	switch filter.Op {
+	case Eq:
 		if filter.Partial {
-			format = matchFmt
-		}
-		match := filter.Match
-		// To allow matches on the backslash itself, the character needs to be replaced first.
-		// Otherwise, it will undo the following replacements.
-		match = strings.ReplaceAll(match, `\`, `\\`)
-		match = strings.ReplaceAll(match, `_`, `\_`)
-		match = strings.ReplaceAll(match, `%`, `\%`)
-		params = append(params, fmt.Sprintf(format, match))
-		if index == len(orFilters.Filters)-1 {
-			continue
+			opString = "LIKE"
+			escapeString = escapeBackslashDirective
+		} else {
+			opString = "="
+		}
+		clause := fmt.Sprintf(`f."%s" %s ?%s`, columnName, opString, escapeString)
+		return clause, []any{formatMatchTarget(filter)}, nil
+	case NotEq:
+		if filter.Partial {
+			opString = "NOT LIKE"
+			escapeString = escapeBackslashDirective
+		} else {
+			opString = "!="
+		}
+		clause := fmt.Sprintf(`f."%s" %s ?%s`, columnName, opString, escapeString)
+		return clause, []any{formatMatchTarget(filter)}, nil
+	case Exists:
+		clause := fmt.Sprintf(`f."%s" IS NOT NULL`, columnName)
+		return clause, []any{}, nil
+	case NotExists:
+		clause := fmt.Sprintf(`f."%s" IS NULL`, columnName)
+		return clause, []any{}, nil
+	case In:
+		fallthrough
+	case NotIn:
+		target := "()"
+		if len(filter.Matches) > 0 {
+			target = fmt.Sprintf("(?%s)", strings.Repeat(", ?", len(filter.Matches)-1))
+		}
+		opString = "IN"
+		if filter.Op == NotIn {
+			opString = "NOT IN"
+		}
+		clause := fmt.Sprintf(`f."%s" %s %s`, columnName, opString, target)
+		matches := make([]any, len(filter.Matches))
+		for i, match := range filter.Matches {
+			matches[i] = match
+		}
+		return clause, matches, nil
+	}
+
+	return "", nil, fmt.Errorf("unrecognized operator: %s", opString)
+}
+
+func (l *ListOptionIndexer) getLabelFilter(filter Filter, dbName string) (string, []any, error) {
+	opString := ""
+	escapeString := ""
+	if len(filter.Field) < 3 || filter.Field[0] != "metadata" || filter.Field[1] != "labels" {
+		return "", nil, fmt.Errorf("expecting a metadata.labels field, got '%s'", strings.Join(filter.Field, "."))
+	}
```

Comment:
```suggestion
```

I feel like this is a bit of defensive programming: `getLabelFilter` is unexporter and only called by `buildORClauseFromFilters`, which uses `isLabelFilter` immediately before the call to check this very condition (well, its negation).

Am I missing anything?

Reply:
Yes, exactly. I believe I wrote the code in `getLabelFilter` first and then filled in the other details. So this was scaffolding that can be deleted.